Source/NSWindow.m

```diff
@@ -4180,6 +4180,17 @@ window that has been ordered out (and thus is logically invisible). We
                 }
               [nc postNotificationName: NSWindowDidMoveNotification
                   object: self];
+              // Check if window moved to an other screen
+              {
+                NSScreen *newScreen = [self _screenForFrame: _frame];
```

Comment:
> 
> 
> It would be better to call [self screen] here as this is already optimised for the common case that the screen stays the same. But you will have to store _screen in a local variable _oldScreen as it will change. Or even better move the notification over to that method. 

Done.

>When we have proper multi monitor support 

What do you mean by "proper"? I think we already have proper multi-monitor support.

> windows may change the screen without being moved, just by rearranging the monitors. This will need to be handled as well.

Agree. I think it should be a separate task/PR.